internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -1901,7 +1911,113 @@ func KexecPrepare(_ runtime.Sequence, data any) (runtime.TaskExecutionFunc, stri
 
 		defer dev.Unlock() //nolint:errcheck
 
-		_, err = grub.ProbeWithCallback(systemDisk.DevPath,
+		bootedEntry, err := sdboot.Probe(systemDisk.DevPath, options.ProbeOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to probe system disk: %w", err)
+		}
+
+		// means we booted using sd-boot
+		if bootedEntry != nil {
+			if err := bootloadermount.PartitionOp(
+				systemDisk.DevPath,
+				[]bootloadermount.Spec{
+					{
+						PartitionLabel: constants.EFIPartitionLabel,
+						FilesystemType: partition.FilesystemTypeVFAT,
+						MountTarget:    constants.EFIMountPoint,
+					},
+				},
+				func() error {
+					var kernelFd int
```

Comment:
I think we need to refactor this a bit, this already becomes a bit of a function that is super hard to read.

we might want to extract these functions out, we can even make them part of the `bootloader` interface

Reply:
:+1: yup, makes sense